pkg/apis/pipeline/v1/taskrun_validation.go

```diff
@@ -116,15 +116,44 @@ func (ts *TaskRunSpec) validateInlineParameters(ctx context.Context) (errs *apis
 	var paramSpec []ParamSpec
 	for _, p := range ts.Params {
 		pSpec := ParamSpec{
-			Name:    p.Name,
-			Default: &p.Value,
+			Name:       p.Name,
+			Default:    &p.Value,
+			Properties: make(map[string]PropertySpec),
+		}
+		if p.Value.ObjectVal != nil {
+			prop := make(map[string]PropertySpec)
+			for k, v := range p.Value.ObjectVal {
+				if fmt.Sprintf("%T", v) == "string" {
+					prop[k] = PropertySpec{Type: ParamTypeString}
+				}
+			}
+			pSpec.Type = ParamTypeObject
+			pSpec.Properties = prop
 		}
 		paramSpec = append(paramSpec, pSpec)
 	}
 	for _, p := range ts.TaskSpec.Params {
 		skip := false
-		for _, ps := range paramSpec {
+		for i, ps := range paramSpec {
 			if ps.Name == p.Name {
+				if p.Default != nil && p.Default.ObjectVal != nil {
+					for k, v := range p.Default.ObjectVal {
+						if paramSpec[i].Default.ObjectVal == nil {
+							paramSpec[i].Default.ObjectVal = map[string]string{k: v}
+						} else {
+							paramSpec[i].Default.ObjectVal[k] = v
+						}
+						if fmt.Sprintf("%T", v) == "string" {
```

Comment:
Can you explain a bit what's going on here or maybe give a yaml example? this is if the literal value of an object param property is "string"? I don't understand why this is a special case or why we're using `sprintf` in this way

Reply:
Thanks for catching that @lbernick! I put it there originally thinking that the object params also allowed us to provide arrays as values. After going through the TEPs regarding object params yesterday, I figured out that we currently only support string values. I missed updating the conditional later. The tests passed because nothing really triggered a failure. I will fix it ASAP.